scripts/quests/otherAreas/An_Affable_Adamantking.lua

```diff
@@ -0,0 +1,321 @@
+                        trade:getGil() == 10000 and
+                        trade:getItemCount() == 3 and
+                        quest:getVar(player, 'Prog') == 1
+                    then
+                        return quest:progressEvent(712)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [710] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1) -- both options return 0
+                end,
+
+                [711] = function(player, csid, option, npc)
+                    if option == 100 then
+                        player:messageSpecial(windurstWoodsID.text.QUEST_CANCELED)
+                        quest:setVar(player, 'Prog', 0) -- reset progress
+                        player:delQuest(quest.areaId, quest.questId)
+                        player:setVar('BstHeadGearQuest_Conquest', 0) -- player can accept other headgear quests upon cancelling without conquest wait
+                    end
+                end,
+
+                [712] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                    quest:setVar(player, 'partsWait', getVanaMidnight())
+                    player:confirmTrade()
+                    player:delGil(10000)
+                    quest:setMustZone(player)
+                end,
+
+                [714] = function(player, csid, option, npc)
+                    if option == 0 then
+                        if player:getGil() > 100000 then
+                            quest:setVar(player, 'partsWait', getVanaMidnight())
+                            player:delGil(100000)
+                            quest:setMustZone(player)
+                        else
+                            player:messageSpecial(windurstWoodsID.text.NOT_HAVE_ENOUGH_GIL)
+                        end
+                    elseif option == 100 then
+                        player:messageSpecial(zones[player:getZoneID()].text.QUEST_CANCELED)
+                        quest:setVar(player, 'Prog', 0)
+                        player:delQuest(quest.areaId, quest.questId)
+                        player:setVar('BstHeadGearQuest_Conquest', 0) -- player can accept other headgear quests upon cancelling
+                    end
+                end,
+
+                [715] = function(player, csid, option, npc)
+                    if
+                        player:getFreeSlotsCount() > 0 and
+                        not player:hasItem(xi.item.QUADAV_PARTS)
+                    then
+                        if quest:getVar(player, 'Prog') == 2 then
+                            quest:setVar(player, 'Prog', 3) -- only progress if current prog is 2, otherwise player is already at, or beyond, Prog 3
+                        end
+
+                        quest:setVar(player, 'partsWait', 0)
+                        npcUtil.giveItem(player, xi.item.QUADAV_PARTS)
```

Comment:
This can fail, and needs to be attempted and checked prior to changing other variables

Reply:
Fixed!